### PR TITLE
[hist] Fix pattern-matching bug in TFormula::HandleLinear()

### PR DESCRIPTION
**Pull request by guitargeek**
Simple C++ code to demonstrate the problem:
```C++
TF1{"f1", "1.0++x++x*x++x*x*x", -1.0, 1.0}.Print();
```
Before this commit, running this line gave you:
```
Formula based function:     f1
        f1 : 1.0++x++x*x++x*x*x Ndim= 1, Npar= 3, Number= 0
 Formula expression:
        ([p0]*(1.0))+([p1]*(x))+([p2]*(x*x))+([p2]*(x*x))*x
```

The problem is that `TFormula::HandleLinear()` uses pattern-matching to
replace the strings representing the linear terms. This is problematic
if one of the strings is a substring of another one.

### CODE REVIEW

**Review by lmoneta — APPROVED** [review]
Thank you for the fix Jonas !
Looks good, I have only a comment, it would be nice to have a new test showing the previous failure in `test/TFormulaParsingTests.h`

File: hist/hist/src/TFormula.cxx [hunk 1]
```diff
@@ -1670,55 +1672,57 @@ void TFormula::HandleExponentiation(TString &formula)
    }
 }
 
+namespace {
+
+// Borrowed from RooHelpers.h, should this go to a central place?
```
Comment (Axel-Naumann) [inline hist/hist/src/TFormula.cxx:1677]: I agree, this should totally go into `core/foundation/inc/ROOT/`. Should this return `vector<string>` or `vector<string_view>`? The latter is dangerous if people do `tokenize("abc cde",...)` but I find that unlikely to happen.

File: hist/hist/src/TFormula.cxx [hunk 2]
```diff
@@ -1670,55 +1672,57 @@ void TFormula::HandleExponentiation(TString &formula)
...
+std::vector<std::string> tokenise(const std::string &str, const std::string &delims, bool returnEmptyToken) {
```
Comment (Axel-Naumann) [inline hist/hist/src/TFormula.cxx:1678]: We generally use US spelling, e.g. `TString::Tokenize`

**Review by Axel-Naumann — APPROVED** [review]
LGTM but see commented. Thanks for fixing this "properly"!

File: hist/hist/src/TFormula.cxx (hunk 2 above)
Comment (guitargeek) [reply hist/hist/src/TFormula.cxx:1678]: Okay, this should be changed then when the function is moved to `core/foundation`.

File: hist/hist/src/TFormula.cxx (hunk 1 above)
Comment (guitargeek) [reply hist/hist/src/TFormula.cxx:1677]: Okay! I don't have time to do this now, but I'll open a GitHub issue about it. Maybe we should have two versions: one returning `vector<string>` and one returning `vector<string_view>`.

**Pull request merged**